sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -174,43 +174,68 @@ private void ApplyAsTarget(Expression expression, Instruction instruction)
 
         private IEnumerable<Instruction> BuildAnnotateCall(SyntaxNode syntaxNode, IMethodSymbol attributeMethodSymbol,
             Expression target) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.Annotate, expressionService.CreateVariable(),
+            BuildFunctionCall(new SyntaxNodeOrToken(syntaxNode), UcfgBuiltInMethodId.Annotate, expressionService.CreateVariable(),
                 expressionService.CreateConstant(attributeMethodSymbol.ToUcfgMethodId()), target);
 
         private IEnumerable<Instruction> BuildAnnotationCall(SyntaxNode syntaxNode, Expression destination, Expression value) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.Annotation, destination, value);
+            BuildFunctionCall(new SyntaxNodeOrToken(syntaxNode), UcfgBuiltInMethodId.Annotation, destination, value);
 
         private IEnumerable<Instruction> BuildArrayGetCall(SyntaxNode syntaxNode, ITypeSymbol nodeTypeSymbol,
             Expression target) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.ArrayGet, expressionService.CreateVariable(), target);
+            BuildFunctionCall(new SyntaxNodeOrToken(syntaxNode), UcfgBuiltInMethodId.ArrayGet,
+                expressionService.CreateVariable(), target);
 
         private IEnumerable<Instruction> BuildArraySetCall(SyntaxNode syntaxNode, Expression destination, Expression value) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.ArraySet, expressionService.CreateVariable(), destination, value);
+            BuildFunctionCall(new SyntaxNodeOrToken(syntaxNode), UcfgBuiltInMethodId.ArraySet,
+                expressionService.CreateVariable(), destination, value);
 
         private IEnumerable<Instruction> BuildConcatCall(SyntaxNode syntaxNode, Expression left, Expression right) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.Concatenation, expressionService.CreateVariable(),
-                left, right);
+            BuildFunctionCall(new SyntaxNodeOrToken(syntaxNode), UcfgBuiltInMethodId.Concatenation,
+                expressionService.CreateVariable(), left, right);
 
-        private IEnumerable<Instruction> BuildEntryPointCall(SyntaxNode syntaxNode, ParameterListSyntax parameterList) =>
-            BuildFunctionCall(syntaxNode, UcfgBuiltInMethodId.EntryPoint, expressionService.CreateVariable(),
-                parameterList.Parameters.Select(expressionService.GetOrDefault).ToArray());
+        private IEnumerable<Instruction> BuildEntryPointCall(BaseMethodDeclarationSyntax baseMethodDeclarationSyntax)
+        {
```

Comment:
This is setting the location on the _AssignCall_ instruction that describes the entry point.
_BasicBlock_ also has a _Location_ property, but this doesn't appear to be set.
Do we need to set _BasicBlock.Location_ too?